src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -52,7 +52,7 @@ public partial class DynamoModel : INotifyPropertyChanged, IDisposable, IEngineC
         private readonly PathManager pathManager;
         private WorkspaceModel currentWorkspace;
         private Timer backupFilesTimer;
-
+        private Dictionary<Guid, string> backupFilesDict = new Dictionary<Guid, string>();
```

Comment:
In this way, the type of BackupFiles in PreferenceSettings has to be changed to a dictionary. I prefer to use list<string>.